### PR TITLE
Add EdgeNetwork Layer

### PR DESCRIPTION
**Pull request by riya-singh28**
## Description

The EdgeNetwork Layer is a submodule designed for message passing in graph neural networks.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by riya-singh28** [comment]
image

**Comment by riya-singh28** [comment]
@rbharath PR is ready for review.

File: docs/source/api_reference/layers.rst [hunk 1]
```diff
@@ -207,6 +207,9 @@ Torch Layers
 .. autoclass:: deepchem.models.torch_models.layers.DTNNEmbedding
   :members:
 
+.. autoclass:: deepchem.models.torch_models.layers.EdgeNetwork
```
Comment (rbharath) [inline docs/source/api_reference/layers.rst:210]: Make sure to add to https://github.com/deepchem/deepchem/blob/master/docs/source/api_reference/torch_layers.csv as well

**Review by rbharath — COMMENTED** [review]
Did a quick pass review. Can you rebase now that the segment_sum PR is merged in and I will do a more thorough review? 

Have one request for a documentation fix

File: docs/source/api_reference/layers.rst (hunk 1 above)
Comment (riya-singh28) [reply docs/source/api_reference/layers.rst:210]: Yes sure, I'll do.

**Review by ARY2260 — APPROVED** [review]
LGTM, @rbharath can you please give a review check

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**